rfcs/0064-documentation-format.md

```diff
@@ -0,0 +1,202 @@
+
+The poll is of the following form:
+- Multiple-choice, allowing people to select all formats they agree with
+- Results are only shown when the poll is closed for it to not be influenced by non-final tallies
+- Answers can be changed while the poll is still active, allowing people to discuss about formats and change their opinion (this is not optional in Discourse)
+- It runs for 1 month to give enough time for less-active people to see it
+- Who voted for which options is made public (Only possible with bar chart in Discourse) TODO: Do we want this or not? Why would we?
+
+## Requirements
+
+- Can be converted to HTML and man pages
+- Inter-file references for being able to link to options from anywhere
+- Ability to create link anchors to most places such that we can link to e.g. paragraphs
+- Widespread editor integration featuring at least highlighting and preferably live-view
+- Good error detection in toolchain and editors, e.g. with a fast and good processor
+- Is decently fast to fully generate, in the range of 10 seconds for the full documentation on an average machine
+- Closure-size of toolchain should be [small](https://github.com/NixOS/nixpkgs/issues/63513).
+- Supports syntax highlighting (with Nix support)
+- Active community supporting the tooling infrastructure
+- Good conversion story from Docbook
+- Can be used and integrated from NixOS option declarations
+
+### Nice-to-have's
+
+- Annotations/links inside code listings for e.g. linking to option docs in `configuration.nix` snippets
+- Ability to make `$ `, `nix-repl>` and other prompts in command line snippets non-copyable
+- Good search integration, e.g. by providing a well-functioning search field
+
+## Format Candidates
+
+All format candidates are listed here with links to demonstrations showing off required features and to get a feel for it. Any tooling can be used to achieve this.
+Ideally this should consist of a standalone repository created for this purpose, not a nixpkgs fork.
+
+| Format | Source | Rendered |
+| --- | --- | --- |
+| Markdown | TODO | TODO | 
+| reST | TODO | TODO | 
+| Asciidoc | TODO | TODO | 
+| DocBook | TODO | TODO | 
+| Texinfo | TODO | TODO | 
+| Nix EDSL | TODO | TODO | 
+
+### Closure sizes of different tools
+
+TODO: Remove this section, as long as tools aren't too big, this isn't relevant or could be showed off in the demos
+For the following comparison NixOS 19.09 is used.
+
+| Name         | Attribute             | Closure size |
+|--------------|-----------------------|--------------|
+| Sphinx       | `python3.pkgs.sphinx` | 195 MB       |
+| Pandoc       | `pandoc`              | 2.4 GB       |
+| Asciidoctor  | `asciidoctor`         | 1.0 GB (on master: 80% smaller)      |
```

Comment:
So if we used 20.03, Asciidoctor's closure would be 80% smaller?

Reply:
Just for a concrete number, I ran `nix path-info -Sh $(nix-build -A asciidoctor)` on `release-20.03`, which output the following:

```
/nix/store/ylb06pz44l6qr9mpr7m9qakc5brvl57a-asciidoctor-2.0.10   320.6M
```

whereas `master` output the following:

```
/nix/store/gbd8nap6fmy078jgijac1m4i8xg72597-asciidoctor-2.0.10   321.3M
```